File: server/src/main/java/vn/elca/ptp/journey/dto/JourneyFilter.java
package vn.elca.ptp.journey.dto;

import java.time.LocalDate;

import vn.elca.ptp.journey.domain.enums.JourneyStatus;

public record JourneyFilter(
    String searchQuery,
    Long countryId,
    Long currencyId,
    JourneyStatus status,
    Long amountFrom,
    Long amountTo,
    LocalDate startDateFrom,
    LocalDate startDateTo,
    LocalDate endDateFrom,
    LocalDate endDateTo) {
}

File: client/src/app/models/journey-filter.ts
export interface JourneyFilter {
  searchQuery?: string;
  countryId?: number;
  currencyId?: number;
  status?: string;
  amountFrom?: number;
  amountTo?: number;
  startDateFrom?: string;
  startDateTo?: string;
  endDateFrom?: string;
  endDateTo?: string;
}
